team/examples/org.eclipse.team.examples.filesystem/src/org/eclipse/team/examples/model/ModelObjectDefinitionFile.java

```diff
@@ -86,24 +86,17 @@ public ModelObjectElementFile[] getModelObjectElementFiles() throws CoreExceptio
 	}
 
 	private static String[] readLines(IStorage file) throws CoreException {
-		BufferedReader reader = new BufferedReader(new InputStreamReader(file.getContents()));
-		String line = null;
-		List<String> result = new ArrayList<>();
-		try {
+		try (BufferedReader reader = new BufferedReader(new InputStreamReader(file.getContents()))) {
```

Comment:
I am not completely sure in this scenario, but I also think this is not behavior preserving: When closing the stream fails, the catch block will throw the `CoreException` and overwrite the `return` statement from the `try` block (unlike before).